posts/copilot-workspace/index.qmd

```diff
@@ -0,0 +1,102 @@
+
+![Screenshot of epidemics package issue 238 which was used as an example application of GitHub Copilot Workspace.](epidemics_issue.png)
+
+### Outcome
+
+GitHub copilot workspace outlined the existing functionality in the package and proposed changes including added a new file called `R/model_sir.R`, which follows the `model_*.R` naming convention used in {epidemics}. As requested, copilot also proposed that the new SIR model would support the use of interventions that have already been implemented in the package.
+
+### Our thoughts
+
+The code generated was an R function that constructed a basic SIR model, and this was well documented using a {roxygen2} documentation block. However, several aspects of the code generated did not match the proposal generated by copilot workspace, and the code contained inadvisable coding practices in R (i.e. code smells). Firstly the SIR model itself. The model implemented follows the standard set of differential equations that are solved using {deSolve}, as requested in our prompt. However, the model does not have any options to input interventions, which the proposed solution by GitHub copilot workspace suggested it would. The second downside is the use of `require(deSolve)` in the function body of the generated code. This is bad practice in R as functions should namespace any functions that are being called from other packages (e.g. `packagename::func_name()`).
+
+This task required copilot workspace to understand the contents and conventions of the {epidemics} R package and generate a simple compartmental model which would integrate with the infrastructure in the package for defining interventions in models . Although it correctly summarised the package contents, its attempt to answer the specific request from our prompt had many inaccuracies. The generated code is likely influenced by the many examples of SIR models written in R on the internet and on GitHub. These might be R scripts or gists and thus coding practises used there are not always appropriate for writing code to be distributed in an R package, resulting in decisions like having `require(deSolve)` in the function.
+
+AI generated code like that seen in this example showcases where users need to be diligent and not assume that the text descriptions from copilot match the code. In the end, this was a test example and was unlikely to be merged into {epidemics} even if the code was correctly, but with the code generated it presents almost no use and for an RSE would require as much time to refactor this SIR model function as it would to write one from scratch following the design principles and demands of the {epidemics} package.
```

Comment:
```suggestion
AI generated code like that seen in this example showcases where users need to be diligent and not assume that the text descriptions from copilot match the code. In the end, this was a test example and was unlikely to be merged into {epidemics} even if the code was correct but with the code generated it presents almost no use and for an RSE would require as much time to refactor this SIR model function as it would to write one from scratch following the design principles and demands of the {epidemics} package.
```